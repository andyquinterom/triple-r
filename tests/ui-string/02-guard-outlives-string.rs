use triple_r::ReusableString;

fn main() {
    let mut guard;
    {
        let mut s = ReusableString::default();
        // The guard borrows `s`, which has a shorter lifetime than `guard`.
        guard = s.recycle();
        // `s` is dropped at the end of this scope.
    }

    // This line should fail to compile because `guard` now holds a dangling
    // reference to the data from `s`, which has been dropped.
    println!("Length: {}", guard.len());
}
