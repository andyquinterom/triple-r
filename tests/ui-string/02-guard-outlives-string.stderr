warning: variable does not need to be mutable
 --> tests/ui-string/02-guard-outlives-string.rs:4:9
  |
4 |     let mut guard;
  |         ----^^^^^
  |         |
  |         help: remove this `mut`
  |
  = note: `#[warn(unused_mut)]` on by default

error[E0597]: `s` does not live long enough
  --> tests/ui-string/02-guard-outlives-string.rs:8:17
   |
6  |         let mut s = ReusableString::default();
   |             ----- binding `s` declared here
7  |         // The guard borrows `s`, which has a shorter lifetime than `guard`.
8  |         guard = s.recycle();
   |                 ^ borrowed value does not live long enough
9  |         // `s` is dropped at the end of this scope.
10 |     }
   |     - `s` dropped here while still borrowed
...
14 |     println!("Length: {}", guard.len());
   |                            ----- borrow later used here
